### PR TITLE
Fix SDK attachContractsMap bug

### PR DESCRIPTION
**Pull request by asaj**
### Description

attachContractsMap should filter out chains for which a complete set of addresses is not present

### Drive-by changes

None

### CODE REVIEW

File: typescript/sdk/src/contracts.ts [hunk 1]
```diff
@@ -53,30 +53,42 @@ function getFactory(
...
 
-export function filterAddresses(
-  addresses: HyperlaneAddresses<any>,
+export function filterAddressesMap(
+  addressesMap: HyperlaneAddressesMap<any>,
   factories: HyperlaneFactories,
-): HyperlaneAddresses<any> {
-  return pick(addresses, Object.keys(factories));
+): HyperlaneAddressesMap<typeof factories> {
+  // Filter out addresses that we do not have factories for
```
Comment (nambrot) [inline typescript/sdk/src/contracts.ts:60]: I don't have full context here, so you might want somebody else to actually review this. IIUC, this is to allow to pass in a superset of addresses and still have attachContracts return the expected contracts shape. Would it be possible to not do this implicitly, but explicitly, i.e. have the caller specify the filtering explicitly?

**Review by nambrot — APPROVED** [review]

File: typescript/sdk/src/contracts.ts [hunk 2]
```diff
@@ -53,30 +53,42 @@ function getFactory(
...
+  const pickedAddressesMap = objMap(addressesMap, (_, addresses) =>
+    pick(addresses, Object.keys(factories)),
```
Comment (jmrossy) [inline typescript/sdk/src/contracts.ts:62]: Nit, hoist Object.keys(factories) to a var as it's used in several places, including in maps

File: typescript/sdk/src/contracts.ts [hunk 3]
```diff
@@ -53,30 +53,42 @@ function getFactory(
...
+  );
+  // Filter out chains for which we do not have a complete set of addresses
+  return objFilter(
+    pickedAddressesMap,
+    (_, addresses): addresses is HyperlaneAddresses<typeof factories> => {
+      return Object.keys(factories)
+        .map((contract) => Object.keys(addresses).includes(contract))
+        .every(Boolean);
```
Comment (jmrossy) [inline typescript/sdk/src/contracts.ts:70]: `Object.keys(addresses).every(a => factoriesAddrs.includes(a));`

**Review by jmrossy — APPROVED** [review]
See optional comments

File: typescript/sdk/src/contracts.ts (hunk 1 above)
Comment (asaj) [reply typescript/sdk/src/contracts.ts:61]: Doing this in a future PR

**Pull request merged**